extern/storage-sealing/input.go

```diff
@@ -311,41 +311,45 @@ func (m *Sealing) SectorAddPieceToAny(ctx context.Context, size abi.UnpaddedPiec
 	}
 
 	m.inputLk.Lock()
-	if _, exist := m.pendingPieces[proposalCID(deal)]; exist {
+	if pp, exist := m.pendingPieces[proposalCID(deal)]; exist {
 		m.inputLk.Unlock()
-		return api.SectorOffset{}, xerrors.Errorf("piece for deal %s already pending", proposalCID(deal))
+		select {
+		case <-pp.doneCh:
+			res := pp.resp.Load().(*pieceAcceptResp)
+			return api.SectorOffset{Sector: res.sn, Offset: res.offset.Padded()}, res.err
+		case <-ctx.Done():
+			return api.SectorOffset{}, ctx.Err()
+		}
 	}
 
-	resCh := make(chan struct {
-		sn     abi.SectorNumber
-		offset abi.UnpaddedPieceSize
-		err    error
-	}, 1)
-
-	m.pendingPieces[proposalCID(deal)] = &pendingPiece{
+	doneCh := make(chan struct{}, 1)
```

Comment:
```suggestion
	doneCh := make(chan struct{})
```

Reply:
Let's keep this for safety, dosen't hurt because the code that writes to this is called sync by the sealer

Reply:
We never really write to this channel, just close it, which doesn't block.